File: scripts/rangers_and_ruffians/RnRItem.py
from .RnRAbility import RnRAbility

class RnRItem:
	def __init__(self, item_def):
		self.name = item_def['name'] 
		self.gold_value = item_def['goldValue'] 
		self.type = item_def['type']
		self.subtype = item_def.get('subtype', '')
		self.ability = RnRAbility(item_def['ability'])
	

    

	def serialize(self):
		serial = dict()
		serial['name'] = self.name 
		serial['goldValue'] = self.gold_value 
		serial['type'] = self.type
		serial['subtype'] = self.subtype
		serial['ability'] = self.ability.serialize()
		
		return serial

File: src/app/data_types/Classes/RnRItem.ts
import { AbilityData } from "../Interfaces/AbilityInterface"
import { ItemData } from "../Interfaces/ItemInterface"
import { RnRAbility } from "./RnRAbility"

export class RnRItem implements ItemData {
    name: string;
    goldValue: number;
    type: string;
    subtype: string;
    ability: AbilityData;
    instantiatedAbility: RnRAbility;

    constructor(data: ItemData) {
        this.name = data.name ?? "Unknown Item";
        this.goldValue = data.goldValue ?? 5;
        this.type = data.type ?? "Item"
        this.subtype = data.type ?? ""
        this.ability = data.ability;
        this.instantiatedAbility = new RnRAbility(data.ability, true, false);
    }
}